scene/main/viewport.cpp

```diff
@@ -2993,17 +2996,31 @@ void Viewport::set_hdr(bool p_hdr) {
 		return;
 	}
 
+	if (p_hdr && (usage == USAGE_2D || usage == USAGE_2D_NO_SAMPLING)) {
+		WARN_PRINT("A Viewport has HDR enabled, but its Usage is set to 2D or 2D No-Sampling. HDR is only supported in Viewports that have their Usage set to 3D or 3D No-Effects. HDR will be disabled for that Viewport.");
+	}
```

Comment:
Maybe the configuration warning is enough as is? Adding debug code like this to the setters seems overkill when we'll already do the check for the editor anyway.

Reply:
Yeah but that's the same for all node configuration warnings which we have currently. Those warnings help discovering the documentation / how the API works, and most users might end up encountering them through the editor first, before attempting advanced code-only scenarios.

So unless this is very critical, a configuration warning + documentation note should be enough (if using the API from code only, one can expect users to read the docs IMO).